MovieApp/MovieApp/App/Core/Combine/Publisher+handleLocalStorage.swift

```diff
@@ -0,0 +1,16 @@
+import Combine
+
+extension Publisher where Output == [MovieDataModel], Failure == RequestError {
+
+    func handleLocalStorage(localDataSource: MovieLocalDataSourceProtocol) -> AnyPublisher<[MovieDataModel], Never> {
+        handleEvents(receiveOutput: {
+            localDataSource.saveLocal(array: $0, category: .topRated)
+        })
+        .replaceError(with: [])
+        .flatMap { _ -> AnyPublisher<[MovieDataModel], Never> in
+            localDataSource.getMoviesPublisher(for: .topRated)
+        }
+        .eraseToAnyPublisher()
+    }
```

Comment:
These types of rx extensions are a great thing and super useful when dealing with generic stuff like error handling or showing prompts. In your case, it would be a better idea to use data source "as-is" - directly calling the data source for the things you want to do.

If I understand correctly you can replace this with a function (in the data source) that does a similar thing and call it a `flatMap`. That way, you'll extract code into a function and make code more readable, which was the intention I presume.

Reply:
Also had to fix `category` parameter when calling `saveLocal` and `getMoviesPublisher`, because I was accidentally saving everything to `.topRated` category.